### DISCUSSION TITLE
Akka.NET v1.4.21: Akka.DependencyInjection changes feedback thread

### DISCUSSION THREAD

**Discussion by Aaronontheweb** [akkadotnet/akka.net#5070] [General]
Please post any feedback or suggestions you have about https://github.com/akkadotnet/akka.net/pull/4814, introduced in Akka.NET v1.4.21, on this thread.

**Comment by wesselkranenborg** [comment]
During upgrading to 1.4.21-beta1 we saw that `IDependencyResolver.GetService<T>()` is returning `object`. Is there a specific reason why it is not directly returning `<T>`?
Now we need to cast it to `T` before we can use the object.

**Reply by Aaronontheweb** [reply]
That sounds like an implementation bug if that's the case - cc @SamEmber 

We can get that patched before v1.4.21 goes stable. THANK YOU for letting us know about this @wesselkranenborg
